hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/utils/FrameUtils.java

```diff
@@ -257,6 +260,32 @@ public static boolean unqualifiedDelegateDetected(final MessageFrame frame) {
         return true;
     }
 
+    /**
+     * Returns true if the given frame is a call to a contract that must be present based on feature flag settings.
+     * @param frame
+     * @param address to check for possible grandfathering
+     * @param featureFlags
+     * @return
+     */
+    public static boolean contractRequired(
+            @NonNull final MessageFrame frame,
+            @NonNull final Address address,
+            @NonNull final FeatureFlags featureFlags) {
+        requireNonNull(frame);
+        requireNonNull(address);
+        requireNonNull(featureFlags);
+
+        Long maybeGrandfatheredNumber = null;
+        if (isLongZero(address)) {
+            try {
+                maybeGrandfatheredNumber = asNumberedContractId(address).contractNum();
+            } catch (final ArithmeticException ignore) {
+                // Not a valid numbered contract id
+            }
+        }
+        return !featureFlags.isAllowCallsToNonContractAccountsEnabled(configOf(frame), maybeGrandfatheredNumber);
```

Comment:
rather unfortunate here, and in `FrameBuilder` above - that the feature flag has the opposite sense of the thing you are really interested in, expressed positively.  Consider a very simple method with a good name that just "reverses" the sense of the feature flag, to wrap usages of the feature flag?

Reply:
I didn't want to put the `!` in the if statements which I would need to if I reverse the meaning of this method